geode-core/src/main/java/org/apache/geode/cache/query/security/RegExMethodAuthorizer.java

```diff
@@ -0,0 +1,169 @@
+  }
+
+  /**
+   * Creates a {@code RegExMethodAuthorizer} object and initializes it so it can be safely used
+   * in a multi-threaded environment.
+   * <p/>
+   *
+   * Applications can use this constructor as part of the initialization for custom authorizers
+   * (see {@link Declarable#initialize(Cache, Properties)}, when using a declarative approach.
+   *
+   * @param cache the {@code Cache} instance that owns this authorizer, required in order to
+   *        configure the default {@link RestrictedMethodAuthorizer}.
+   * @param allowedPatterns the regular expressions that will be used to determine whether a method
+   *        is authorized or not.
+   */
+  public RegExMethodAuthorizer(Cache cache, Set<String> allowedPatterns) {
+    Objects.requireNonNull(cache, NULL_CACHE_MESSAGE);
+    Objects.requireNonNull(allowedPatterns, NULL_REGULAR_EXPRESSIONS_MESSAGE);
+
+    this.allowedPatterns = Collections.unmodifiableSet(allowedPatterns);
+    this.compiledPatterns = Collections.unmodifiableSet(compilePatterns());
+    this.restrictedMethodAuthorizer = new RestrictedMethodAuthorizer(cache);
+  }
+
+  /**
+   * Creates a {@code RegExMethodAuthorizer} object and initializes it so it can be safely used
+   * in a multi-threaded environment.
+   * <p/>
+   *
+   * @param restrictedMethodAuthorizer the default {@code RestrictedMethodAuthorizer} to use.
+   * @param allowedPatterns the regular expressions that will be used to determine whether a method
+   *        is authorized or not.
+   */
+  public RegExMethodAuthorizer(RestrictedMethodAuthorizer restrictedMethodAuthorizer,
+      Set<String> allowedPatterns) {
+    Objects.requireNonNull(allowedPatterns, NULL_REGULAR_EXPRESSIONS_MESSAGE);
+    Objects.requireNonNull(restrictedMethodAuthorizer, NULL_AUTHORIZER_MESSAGE);
+
+    this.allowedPatterns = Collections.unmodifiableSet(allowedPatterns);
+    this.compiledPatterns = Collections.unmodifiableSet(compilePatterns());
+    this.restrictedMethodAuthorizer = restrictedMethodAuthorizer;
+  }
+
+  /**
+   * Executes the authorization logic to determine whether the {@code method} is allowed to be
+   * executed on the {@code target} object instance.
+   * If the {@code target} object is an instance of {@link Region}, this methods also ensures that
+   * the user has the {@code DATA:READ} permission granted for the target {@link Region}.
+   * <p/>
+   *
+   * @param method the {@link Method} that should be authorized.
+   * @param target the {@link Object} on which the {@link Method} will be executed.
+   * @return {@code true} if the {@code method} can be executed on on the {@code target} instance,
+   *         {@code false} otherwise.
+   *
+   * @see org.apache.geode.cache.query.security.MethodInvocationAuthorizer
+   */
+  @Override
+  public boolean authorize(Method method, Object target) {
+    // Return false for known dangerous methods.
+    if (restrictedMethodAuthorizer.isKnownDangerousMethod(method, target)) {
```

Comment:
Do you feel like changing the wording "dangerous" to something like isAlwaysRestrictedMethod() or isPermanentlyRestrictedMethod() will be more informative?

Reply:
Hey @nabarunnag, thanks for reviewing this 👍 .
Regarding the method name change, it's of course doable but not as part of this PR: `isKnownDangerousMethod` belongs to the `RestrictedMethodAuthorizer` class and it's also used by the other 3 authorizers, so it's better to create a new GEODE JIRA for that.